README.md

```diff
@@ -1,26 +1,227 @@
+
+Regardless of transport, every RPC has some common properties: caller name,
+service name, procedure name, encoding name, deadline or TTL, headers, baggage
+(multi-hop headers), and tracing.
+Each RPC can also have an optional shard key, routing key, or routing delegate
+for advanced routing.
+YARPC transports use a shared API for capturing RPC metadata, so middleware can
+apply to requests over any transport.
+
+Each YARPC transport protocol can implement inbound handlers and outbound
+callers. Each of these can support different RPC types, like unary (request and
+response) or oneway (request and receipt) RPC. A future release of YARPC will
+add support for other RPC types including variations on streaming and pubsub.
+
 
 ## Installation
 
 ```
 go get -u go.uber.org/yarpc
 ```
 
-If using [Glide](https://github.com/Masterminds/glide), *at least* `glide version 0.12` is required to install:
+If using [Glide](https://github.com/Masterminds/glide), *at least* `glide
+version 0.12` is required to install:
 
 ```
 $ glide --version
 glide version 0.12.2
 
-$ glide get 'go.uber.org/yarpc#~0.5'
+$ glide get 'go.uber.org/yarpc#^1'
+```
+
+
+## Examples
+
+This example illustrates a simple service that implements a handler for a
+`Hello::echo` Thrift procedure.
+
+```thrift
+service Hello {
+    EchoResponse echo(1:EchoRequest echo)
+}
+
+struct EchoRequest {
+    1: required string message;
+    2: required i16 count;
+}
+
+struct EchoResponse {
+    1: required string message;
+    2: required i16 count;
+}
+```
+
+A go:generate directive informs `go generate` how to produce the Thrift models
+and YARPC bindings for the echo service.
+
+```go
+//go:generate thriftrw --plugin=yarpc echo.thrift
+```
+
+Setting up a YARPC dispatcher configures inbounds and outbounds for supported
+transport protocols and RPC types.
+This sets a service up to receive HTTP requests on port 8080 and send requests
+to itself.
+YARPC funnels requests from all inbound transports into its routing table, and
+organizes outbounds by name.
+
+```go
+httpTransport := http.NewTransport()
+dispatcher := yarpc.NewDispatcher(yarpc.Config{
+    Name: "hello",
+    Inbounds: yarpc.Inbounds{
+        httpTransport.NewInbound(":8080"),
+    },
+    Outbounds: yarpc.Outbounds{
+        "hello": {
+            Unary: httpTransport.NewSingleOutbound("http://127.0.0.1:8080"),
+        },
+    },
+})
+```
+
+The dispatcher governs the lifecycle of every inbound, outbound, and the
+singleton for each transport protocol.
+The singleton can manage the lifecycle of shared peers and connections.
+
+```go
+if err := dispatcher.Start(); err != nil {
+    log.Fatal(err)
+}
+defer dispatcher.Stop()
+```
+
+At the end of `main`, we block until we receive a signal to exit, then unravel
+anything deferred like `dispatcher.Stop()`, shutting down gracefully.
+
 ```
+signals := make(chan os.Signal, 1)
+signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+<-signals
```

Comment:
That's probably true. Would providing a small standalone function be better? It could even be something as opaque as `yarpc.WaitForShutdownSignal()`, which completely hides the channel.